tiledb/api/src/array/dimension/mod.rs

```diff
@@ -207,62 +209,63 @@ impl<'ctx> Builder<'ctx> {
         })
     }
 
-    /// N.B., This API will eventually be removed when we fixup the current
-    /// implementation of `new` to take a `DimensionConstraint` argument instead
-    /// of the serde values for range and extent.
-    pub fn new_string(
+    pub fn new_optional(
         context: &'ctx Context,
         name: &str,
         datatype: Datatype,
+        domain: Option<SingleValueRange>,
```

Comment:
I don't think this is sound without a safeguard that `domain` and `extent` are the same physical type.

Reply:
Its absolutely not. Though I swear I had type checks in here for this specific reason. I was probably distracted by something shiny and put them in the wrong place or something.